repose-aggregator/components/filters/munging-filter/src/main/resources/META-INF/schema/config/munging.xsd

```diff
@@ -0,0 +1,176 @@
+           xmlns:vc="http://www.w3.org/2007/XMLSchema-versioning"
+           xmlns:xerces="http://xerces.apache.org"
+           xmlns:saxon="http://saxon.sf.net/"
+           elementFormDefault="qualified"
+           targetNamespace="http://docs.openrepose.org/repose/munging/v1.0"
+           xmlns="http://docs.openrepose.org/repose/munging/v1.0">
+
+    <!-- Elements -->
+    <xs:element name="munging" type="MungingConfig"/>
+
+    <!-- Types -->
+    <xs:complexType name="MungingConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    The root config type for the Munging filter configuration file.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:sequence>
+            <xs:element name="change" type="ChangeDetails" minOccurs="1" maxOccurs="unbounded"/>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="ChangeDetails">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    The set of patches to be applied for a given url path.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:sequence>
+            <xs:element name="header-filter" type="HeaderFilter" minOccurs="0" maxOccurs="1"/>
+
+            <xs:element name="request" type="Patch" minOccurs="0" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            The changes to be applied to the incoming request.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+
+            <xs:element name="response" type="Patch" minOccurs="0" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            The changes to be applied to the outbound response.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+        </xs:sequence>
+
+        <xs:attribute name="path" type="xs:string" use="optional" default=".*">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        A regex of the request paths this set of changes should be applied to.
+                        Defaults to matching on everything.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+
+        <xs:assert vc:minVersion="1.1"
+                   xpathDefaultNamespace="##targetNamespace"
+                   test="request or response"
+                   xerces:message="You must specify request and/or response behavior"
+                   saxon:message="You must specify request and/or response behavior">
+        </xs:assert>
```

Comment:
I think there's a way to get the same behavior without relying on an assert. Something like...

```
<xs:sequence>
    <xs:choice>
        <xs:sequence>
            <xs:element name="request" type="Patch" minOccurs="1" maxOccurs="1"/>
            <xs:element name="response" type="Patch" minOccurs="0" maxOccurs="1"/>
        </xs:sequence>
        <xs:element name="response" type="Patch" minOccurs="1" maxOccurs="1"/>
    </xs:choice>
</xs:sequence>
```

I made the minOccurs and maxOccurs explicit, but in some places, they don't need to be. I didn't want to pull up the docs to figure out what places, though.